dali/c_api/c_api.cc

```diff
@@ -138,11 +140,14 @@ void SetExternalInputTensors(daliPipelineHandle *pipe_handle, const char *name,
   else
     order = AccessOrder::host();
 
+  int device_id = order.is_device() ? order.device_id() : dali::CPU_ONLY_DEVICE_ID;
```

Comment:
I guess `device_id` should make it to the C API, too.

Reply:
Merged already as a separate PR :)